cmd/influxd/run.go

```diff
@@ -128,35 +106,92 @@ func parseConfig(path, hostname string) *Config {
 	return config
 }
 
-// creates and initializes a broker at a given path.
-func openBroker(path, addr string) *messaging.Broker {
+// creates and initializes a broker.
+func openBroker(path string, u *url.URL, initializing bool, joinURLs []*url.URL) *messaging.Broker {
+	// Ignore if there's no existing broker and we're not initializing or joining.
+	if !fileExists(path) && !initializing && len(joinURLs) == 0 {
+		return nil
+	}
+
+	// Create broker.
 	b := messaging.NewBroker()
-	if err := b.Open(path, addr); err != nil {
+	if err := b.Open(path, u); err != nil {
 		log.Fatalf("failed to open broker: %s", err)
 	}
+
+	// If this is a new broker then we can initialie two ways:
+	//   1) Start a brand new cluster.
+	//   2) Join an existing cluster.
+	if initializing {
+		if len(joinURLs) == 0 {
+			initializeBroker(b)
+		} else {
+			joinBroker(b, joinURLs)
+		}
+	}
+
 	return b
 }
 
-// creates and initializes a server at a given path.
-func openServer(path string) *influxdb.Server {
+// initializes a new broker.
+func initializeBroker(b *messaging.Broker) {
```

Comment:
This function seems hardly worth the trouble. :-) Purely a style concern, just curious.